src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -685,8 +687,6 @@ def validate_plan_switch_compatibility(cmd, client, src_functionapp_instance, de
 
 def set_functionapp(cmd, resource_group_name, name, **kwargs):
     instance = kwargs['parameters']
-    if not instance or 'function' not in instance.kind:
-        raise ValidationError('Not a function app to update')
```

Comment:
Any reason we moved this check was removed from set_functionapp?

Reply:
`az functionapp update` uses `generic_update_command` which runs getter, custom_func, and setter (in that order). I've moved the check from setter to custom_func in this commit, and in a later commit I've created a custom getter and moved the check there. Having the check earlier allows us to give a more detailed error message and exit execution earlier in case of invalid arguments.